addon/services/file-manager.js

```diff
@@ -136,30 +269,46 @@ export default Ember.Service.extend({
         return new Ember.RSVP.Promise((resolve, reject) => {
             Ember.$.ajax(url, {
                 method,
-                data,
+                data: options.data,
                 processData: false,
                 headers: {
                     Authorization: `Bearer ${accessToken}`
                 },
-                success: (data) => resolve(data),
-                error: (_, __, error) => reject(error)
+                error: (_, __, error) => reject(error),
+                success: (data) => resolve(data)
             });
         });
     },
 
-    _pushToStore(data, id, parentFolder) {
-        // Hack the Waterbutler entity to look like the file Ember expects
-        data.data.id = id;
-        data.data.type = 'file';
-        let attr = data.data.attributes;
-        attr.materializedPath = attr.materialized;
-        attr.dateModified = attr.modified;
-
-        let newFolder = this.get('store').push(data);
+    /**
+     * Force-reload a model from the API.
+     *
+     * @method _reloadModel
+     * @private
+     * @param {Object} model Any Ember model with a `reload` method.
+     * @return {Promise} Promise that resolves to the reloaded model or
+     * rejects with an error message.
+     */
+    _reloadModel(model) {
+        this.set('isReloading', true);
+        return model.reload().then((freshModel) => {
+            this.set('isReloading', false);
```

Comment:
And on a more positive tone: I really will try to think of alternatives when I pick this up- and would be happy to sit down and discuss logic if you want to brainstorm today. Just let me know what I can do to help.